cvat-ui/src/components/annotation-page/canvas/views/context-image/context-image.tsx

```diff
@@ -0,0 +1,119 @@
+// Copyright (C) 2023 CVAT.ai Corporation
+//
+// SPDX-License-Identifier: MIT
+
+import './styles.scss';
+import React, { useEffect, useRef, useState } from 'react';
+import { useSelector } from 'react-redux';
+import PropTypes from 'prop-types';
+import notification from 'antd/lib/notification';
+import Spin from 'antd/lib/spin';
+import Text from 'antd/lib/typography/Text';
+import { SettingOutlined } from '@ant-design/icons';
+
+import { CombinedState } from 'reducers';
+import ContextImageSelector from './context-image-selector';
+
+interface Props {
+    offset: number[];
+}
+
+function ContextImage(props: Props): JSX.Element {
+    const { offset } = props;
+    const canvasRef = useRef<HTMLCanvasElement>(null);
+    const job = useSelector((state: CombinedState) => state.annotation.job.instance);
+    const { number: frame, relatedFiles } = useSelector((state: CombinedState) => state.annotation.player.frame);
+    const frameIndex = frame + (offset[0] || 0);
+
+    const [contextImageData, setContextImageData] = useState<Record<string, ImageBitmap>>({});
+    const [fetching, setFetching] = useState<boolean>(false);
+    const [contextImageOffset, setContextImageOffset] = useState<number>(
+        Math.min(offset[1] || 0, relatedFiles),
```

Comment:
I think we need to use named variables for offset[0]/offset[1] for clarity

Reply:
What do you mean? They have named variables like ``frameIndex`` and ``contextImageOffset``. Or do you mean something else?

Reply:
I mean something like 
`const defaultFrameOffset = (offset[0] || 0);` 
`const defaultContextImageOffset = (offset[1] || 0);`